browser/src/Services/VersionControl/VersionControlView.tsx

```diff
@@ -0,0 +1,224 @@
+    children?: React.ReactNode
+}
+
+interface State {
+    modified: boolean
+    staged: boolean
+    untracked: boolean
+}
+
+class VersionControlView extends React.Component<IProps, State> {
+    public state: State = {
+        modified: true,
+        staged: true,
+        untracked: true,
+    }
+
+    public async componentDidMount() {
+        await this.props.getStatus()
+    }
+
+    public async componentDidCatch(e: Error) {
+        this.props.setError(e)
+    }
+
+    public toggleVisibility = (section: keyof State) => {
+        this.setState(prevState => ({ ...prevState, [section]: !prevState[section] }))
+    }
+
+    public toggleOrAction = (id: string) => {
+        if (id === "modified" || id === "staged" || id === "untracked") {
+            this.toggleVisibility(id)
+        }
+        this.props.handleSelection(id)
+    }
+
+    public insertIf(condition: boolean, element: string[]) {
+        return condition ? element : []
+    }
+
+    public render() {
+        const error = this.props.hasError && "Something Went Wrong!"
+        const inactive = !this.props.activated && "Version Control Not Available"
+        const warning = error || inactive
+        const { modified, staged, untracked } = this.props.status
+        const ids = [
+            "modified",
+            ...this.insertIf(this.state.modified, modified),
+            "staged",
+            ...this.insertIf(this.state.staged, staged),
+            "untracked",
+            ...this.insertIf(this.state.untracked, untracked),
+        ]
+
+        return warning ? (
+            <SectionTitle>
+                <Title>{warning}</Title>
+            </SectionTitle>
+        ) : (
+            <VimNavigator
+                ids={ids}
+                active={this.props.hasFocus}
+                onSelected={this.toggleOrAction}
+                render={selectedId => (
+                    <StatusContainer>
+                        <GitStatus
+                            symbol="M"
+                            files={modified}
+                            titleId="modified"
+                            selectedId={selectedId}
+                            visibility={this.state.modified}
+                            onClick={this.props.handleSelection}
+                            toggleVisibility={() => this.toggleVisibility("modified")}
+                        />
+                        <GitStatus
+                            symbol="S"
+                            titleId="staged"
+                            files={staged}
+                            selectedId={selectedId}
+                            visibility={this.state.staged}
+                            onClick={this.props.handleSelection}
+                            toggleVisibility={() => this.toggleVisibility("staged")}
+                        />
+                        <GitStatus
+                            files={untracked}
+                            symbol="?"
```

Comment:
@CrossR thanks for the feedback, i've run it locally intermittently for a while so definitely don't have fresh eyes anymore, I could do a quick replacement with another FA icon and see how that looks, re staged and unstaged i'd be up for changing that but I think a little later only because it'll require a bit of work to remove that since I've structured it with that particular separation in mind